### PR TITLE
[SYCL] Don't create a dependency on a zero-sized accessor

### PR DESCRIPTION
**Pull request by aelovikov-intel**
No memory transfer is needed in this case. Fixes failure on the existing
`Basic/accessor/empty_zero_dim_accessor.cpp` on some platforms.

Needs #10780 to be merged first.

### CODE REVIEW

File: sycl/source/handler.cpp [hunk 1]
```diff
@@ -499,7 +499,8 @@ void handler::associateWithHandlerCommon(detail::AccessorImplPtr AccImpl,
                                          int AccTarget) {
   detail::Requirement *Req = AccImpl.get();
   // Add accessor to the list of requirements.
-  CGData.MRequirements.push_back(Req);
+  if (Req->MAccessRange.size() != 0)
```
Comment (maarquitos14) [inline sycl/source/handler.cpp:502]: Just to make sure I understand correctly, this is the only real change in this PR, right? The rest comes from #10780, right?

**Review by maarquitos14 — COMMENTED** [review]
Is there anyway we can test this? Maybe a unit test? I'm not sure.

File: sycl/source/handler.cpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/source/handler.cpp:502]: Yes. One can look into the last commit in this PR to limit the diff to the actual change. I plan to mark the PR as ready once I rebase on top of origin/sycl with your PR merged.

**Comment by aelovikov-intel** [comment]
> Is there anyway we can test this? Maybe a unit test? I'm not sure.

We already have a failing e2e test for this, it just requires HW that we only have internally.

**Review by maarquitos14 — APPROVED** [review]

**Comment by aelovikov-intel** [comment]
Ping @sergey-semenov

**Review by sergey-semenov — APPROVED** [review]

**Pull request merged**